src/hera/workflows/container_set.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import annotations
+
+from typing import List, Optional, Union
+
+from hera.workflows._mixins import (
+    ContainerMixin,
+    EnvMixin,
+    IOMixin,
+    ResourceMixin,
+    TemplateMixin,
+    VolumeMountMixin,
+)
+from hera.workflows.container import Container
+from hera.workflows.models import (
+    ContainerNode,
+    ContainerSetRetryStrategy,
+    ContainerSetTemplate as _ModelContainerSetTemplate,
+    Template as _ModelTemplate,
+    VolumeMount,
+)
+
+
+class ContainerSet(
+    IOMixin,
+    ContainerMixin,
+    EnvMixin,
+    TemplateMixin,
+    ResourceMixin,
+    VolumeMountMixin,
+):
+    containers: List[Union[Container, ContainerNode]]
+    container_set_retry_strategy: Optional[ContainerSetRetryStrategy] = None
```

Comment:
@samj1912 and @elliotgunton this is required because of the conflict with `retry_strategy` on the `TemplateMixin`. I figured adding a prefix here is better than adjusting the template mixing since that will push a `template_retry_strategy` downstream to all inheriting classes. This way the "specialized" template class sets the field it needs

Reply:
Sensible! 💯